insights/parsers/dmesg.py

```diff
@@ -59,42 +63,57 @@ def has_startswith(self, prefix):
             for line in self.lines
         )
 
-    def get_after(self, timestamp, lines=None):
+    def get_after(self, timestamp, s=None):
         """
-        Get a list of lines after the given time stamp.
-
-        Note that the time stamp is the floating point number of seconds
-        after the boot time, and is not related to an actual datetime.  If
-        a time stamp is not found on the line between square brackets, then
-        it is treated as a continuation of the previous line and is only
-        included if the previous line's timestamp is greater than the
-        timestamp given.  Because continuation lines are only included if a
-        previous line has matched, this means that searching in logs that do
-        not have a time stamp produces no lines.
+        Find all the (available) logs that are after the given time stamp.
+
+        If `s` is not supplied, then all lines are used.  Otherwise, only the
+        lines contain the `s` are used.  `s` can be either a single string or a
+        strings list. For list, all keywords in the list must be found in the
+        line.
+
+        .. note::
+            The time stamp is the floating point number of seconds
+            after the boot time, and is not related to an actual datetime.  If
+            a time stamp is not found on the line between square brackets, then
+            it is treated as a continuation of the previous line and is only
+            included if the previous line's timestamp is greater than the
+            timestamp given.  Because continuation lines are only included if a
+            previous line has matched, this means that searching in logs that do
+            not have a time stamp produces no lines.
 
         Parameters:
             timestamp(float): log lines after this time are returned.
-            lines(list): the list of log lines to search (e.g. from ``get``).
+            s(str or list): one or more strings to search for.
                 If not supplied, all available lines are searched.
 
         Yields:
             Log lines with time stamps after the given time.
         """
-        if lines is None:
-            lines = self.lines
+        if not isinstance(timestamp, float):
+            raise ParseException(
```

Comment:
@xiangce should this be a `TypeError` instead since it is called by a rule after the data has been parsed?

Reply:
`TypeError` makes more sense, will update it in next commit. Thanks.